packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,79 +9,153 @@ import 'package:file/file.dart';
 import 'package:file/local.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:platform/platform.dart';
 
 import 'package:flutter_goldens_client/client.dart';
+import 'package:flutter_goldens_client/skia_client.dart';
+
 export 'package:flutter_goldens_client/client.dart';
+export 'package:flutter_goldens_client/skia_client.dart';
 
 /// Main method that can be used in a `flutter_test_config.dart` file to set
 /// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
-/// works for the current test.
+/// works for the current test. _Which_ FlutterGoldenFileComparator is
+/// instantiated is based on the current testing environment.
 Future<void> main(FutureOr<void> testMain()) async {
-  goldenFileComparator = await FlutterGoldenFileComparator.fromDefaultComparator();
+  const Platform platform = LocalPlatform();
+  goldenFileComparator = _testingWithSkiaGold(platform)
+    ? await FlutterSkiaGoldFileComparator.fromDefaultComparator()
```

Comment:
Nested ?: are kinda awkward and hard to read. Maybe use at least one real `if` to make this easier to read?